src/main/java/org/broadinstitute/dsm/jobs/UPSTrackingJob.java

```diff
@@ -164,65 +165,59 @@ private static void updateTrackingInfo(String statusType,
-                    if (kitDDPNotification != null) {
-                        logger.info("Triggering DDP for kit going to participant with external order number: " + kit.getExternalOrderNumber());
-                        EventUtil.triggerDDP(kitDDPNotification);
-                    }
-                    else {
-                        logger.error("delivered kitDDPNotification was null for " + kit.getExternalOrderNumber());
-                    }
+        } else { // this is a return
+            if (earliestInTransitTime != null && !kit.isCEOrdered()) {
+                // if we have the first date of an inbound event, create an order in CE
+                // using the earliest date of inbound event
+                OrderResponse orderResponse = orderRegistrar.orderTest(DSMServer.careEvolveAuth, kit.getHRUID(), kit.getKitLabel(), kit.getExternalOrderNumber(), earliestInTransitTime);
+                if (orderResponse.hasError()) {
+                    throw new RuntimeException("Could not place CareEvolve order for " + kit.getKitLabel() + " due to " + orderResponse.getError());
                 }
 
-            }
-            else { // this is a return
-                if (earliestInTransitTime != null && !kit.isCEOrdered()) {
-                    // if we have the first date of an inbound event, create an order in CE
-                    // using the earliest date of inbound event
-                    orderRegistrar.orderTest(DSMServer.careEvolveAuth, kit.getHRUID(), kit.getKitLabel(), kit.getExternalOrderNumber(), earliestInTransitTime);
-                    logger.info("Placed CE order for kit with external order number " + kit.getExternalOrderNumber());
-                    kit.changeCEOrdered(true);
-                } else {
-                    logger.info("No return events for " + kit.getKitLabel() + ".  Will not place order yet.");
-                }
-                if (shouldTriggerEventForReturnKitDelivery(statusType, oldType)) {
-                    KitUtil.setKitReceived(kit.getKitLabel());
-                    logger.info("RECEIVED: " + trackingId);
-                    KitDDPNotification kitDDPNotification = KitDDPNotification.getKitDDPNotification(SQL_SELECT_KIT_FOR_NOTIFICATION_EXTERNAL_SHIPPER + SELECT_BY_RETURN_NUMBER, new String[] { RECEIVED, trackingId }, 2);//todo change this to the number of subkits but for now 2 for test boston works
-                    if (kitDDPNotification != null) {
-                        logger.info("Triggering DDP for received kit with external order number: " + kit.getExternalOrderNumber());
-                        EventUtil.triggerDDP(kitDDPNotification);
+                logger.info("Placed CE order for kit " + kit.getKitLabel() + " with external order number " + kit.getExternalOrderNumber() + " and CE ack" + orderResponse.getHandle());
 
+                TransactionWrapper.inTransaction(conn -> {
```

Comment:
The new approach here is to bundle some db updates into a single transaction here if CE ordering completes.  Birch is already doing de-dup, so if a subsequent job incorrectly attempts to place a redundant order, Birch will ignore it.